includes/load-yourls.php

```diff
@@ -1,200 +1,26 @@
-require_once( YOURLS_INC.'/functions-infos.php' );
-require_once( YOURLS_INC.'/functions-deprecated.php' );
-
-// Load auth functions if needed
-if( yourls_is_private() ) {
-	require_once( YOURLS_INC.'/functions-auth.php' );
-}
-
-// Enforce UTC timezone to suppress PHP warnings -- correct date/time will be managed using the config time offset
-date_default_timezone_set( 'UTC' );
-
-// Load locale
-yourls_load_default_textdomain();
-
-// Check if we are in maintenance mode - if yes, it will die here.
-yourls_check_maintenance_mode();
-
-// Fix REQUEST_URI for IIS
-yourls_fix_request_uri();
-
-// If request for an admin page is http:// and SSL is required, redirect
-if( yourls_is_admin() && yourls_needs_ssl() && !yourls_is_ssl() ) {
-	if ( 0 === strpos( $_SERVER['REQUEST_URI'], 'http' ) ) {
-		yourls_redirect( preg_replace( '|^http://|', 'https://', $_SERVER['REQUEST_URI'] ) );
-		exit();
-	} else {
-		yourls_redirect( 'https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'] );
-		exit();
-	}
-}
-
-// Create the YOURLS object $ydb that will contain everything we globally need
-global $ydb;
-
-// Allow drop-in replacement for the DB engine
-if( file_exists( YOURLS_USERDIR.'/db.php' ) ) {
-	require_once( YOURLS_USERDIR.'/db.php' );
-} else {
-	require_once( YOURLS_INC.'/class-mysql.php' );
-	yourls_db_connect();
-}
-
-// Allow early inclusion of a cache layer
-if( file_exists( YOURLS_USERDIR.'/cache.php' ) )
-	require_once( YOURLS_USERDIR.'/cache.php' );
-
-// Abort initialization here if fast init wanted (for tests/debug/do not use)
-if (defined('YOURLS_FAST_INIT') && YOURLS_FAST_INIT){
-    return;
-}
-
-// Read options right from start
-yourls_get_all_options();
-
-// Register shutdown function
-register_shutdown_function( 'yourls_shutdown' );
-
-// Core now loaded
-yourls_do_action( 'init' ); // plugins can't see this, not loaded yet
-
-// Check if need to redirect to install procedure
-if( !yourls_is_installed() && !yourls_is_installing() ) {
-	yourls_redirect( yourls_admin_url( 'install.php' ), 302 );
-}
-
-// Check if upgrade is needed (bypassed if upgrading or installing)
-if ( !yourls_is_upgrading() && !yourls_is_installing() ) {
-	if ( yourls_upgrade_is_needed() ) {
-		yourls_redirect( YOURLS_SITE .'/admin/upgrade.php', 302 );
-	}
-}
+require __DIR__ . '/vendor/autoload.php';
 
-// Init all plugins
-yourls_load_plugins();
-yourls_do_action( 'plugins_loaded' );
+// Set up YOURLS config
 
-// Is there a new version of YOURLS ?
-if( yourls_is_installed() && !yourls_is_upgrading() ) {
-    yourls_tell_if_new_version();
+$config = new \YOURLS\Config\Config;
+/* The following require has to be at global level so the variables inside config.php, including user defined if any,
+ * are registered in the global scope. If this require is moved in \YOURLS\Config\Config, $yourls_user_passwords for
+ * instance isn't registered.
+ */
+if (!defined('YOURLS_CONFIGFILE')) {
+    define('YOURLS_CONFIGFILE', $config->find_config());
```

Comment:
Yes because it's used in other places (such as `includes/auth.php`)
But at the same time, no, because constants make tests difficult so I'd rather replace most of them with function calls.
The problem here is backward compat because some const are widely used.

Reply:
> The problem here is backward compat because some const are widely used.

Ah, right.